astropy/io/fits/convenience.py

```diff
@@ -709,6 +712,110 @@ def info(filename, output=None, **kwargs):
+        by examples.
+
+        By extension number::
+
+            quickdiff('inA.fits', 'inB.fits', 0)      # the primary HDU
+            quickdiff('inA.fits', 'inB.fits', 2)      # the second extension
+            quickdiff('inA.fits', 'inB.fits', ext=2)  # the second extension
+
+        By name, i.e., ``EXTNAME`` value (if unique)::
+
+            quickdiff('inA.fits', 'inB.fits', 'sci')
+            quickdiff('inA.fits', 'inB.fits', extname='sci')  # equivalent
+
+        Note ``EXTNAME`` values are not case sensitive
+
+        By combination of ``EXTNAME`` and ``EXTVER`` as separate
+        By combination of ``EXTNAME`` and ``EXTVER`` as separate
+        arguments or as a tuple::
+
+            quickdiff('inA.fits', 'inB.fits', 'sci', 2)  # EXTNAME='SCI' & EXTVER=2
+            quickdiff('inA.fits', 'inB.fits', extname='sci', extver=2)  # equivalent
+            quickdiff('inA.fits', 'inB.fits', ('sci', 2))  # equivalent
+
+        Ambiguous or conflicting specifications will raise an exception::
+
+            quickdiff('inA.fits', 'inB.fits', ext=('sci',1), extname='err', extver=2)
+
+
+    kwargs
+        Any additional keyword arguments to be passed to
+        `astropy.io.fits.FITSDiff`.
+
+    Notes
+    -----
+    The primary use for the `quickdiff` function is to allow quick print out
+    of a FITS difference report and will write to ``sys.stdout``.
+    To save the diff report to a file please use `astropy.io.fits.FITSDiff`
+    directly.
+    """
+
+    if isinstance(inputa, string_types) and isinstance(inputb, string_types):
+
+        # Use ext keywords if filename strings provided, unless none exist, then run FITSDIFF directly
+        if len(args) == 0 and 'ext' not in kwargs and 'extname' not in kwargs and 'extver' not in kwargs:
+            print(FITSDiff(inputa, inputb, **kwargs).report())
+
+        else:
+            # Use handy _getext to interpret any ext keywords
+            modea, closeda = _get_file_mode(inputa)
+            modeb, closedb = _get_file_mode(inputb)
+            hdulista, extidxa = _getext(inputa, modea, *args, **kwargs)
+            hdulistb, extidxb = _getext(inputa, modeb, *args, **kwargs)
+
+            # Pop those extension keywords if they exist
+            kwargs.pop('ext', None)
+            kwargs.pop('extname', None)
+            kwargs.pop('extver', None)
+
+            try:
+                hdua = hdulista[extidxa]
+                hdub = hdulistb[extidxb]
+                print(HDUDiff(hdua, hdub, **kwargs).report())
+
+            finally:
+                # See note about _close in getheader function
+                hdulista._close(closed=closeda)
+                hdulistb._close(closed=closedb)
+
+    # If input is not a string, can feed HDU objects directly, but won't
+    # take extension keywords
+    else:
+        if len(args) != 0 or 'ext' in kwargs or 'extname' in kwargs or 'extver' in kwargs:
```

Comment:
> I'm guessing since you put it there it's an astropy approved design?

Well it's some sort of Python design called ["EAFP" or "Easier to ask for forgiveness than permission"](https://docs.python.org/3/glossary.html#term-eafp). But I'm just a deputy (no subproject-lead) without commit-rights so the "just because I said so" rule doesn't apply here. It was just a suggestion because it would simplify handling `HDUList`, `strings`, `HDU`s and file-like-objects without explicitly checking for them.